extensions/aegis/src/component/otp/otp-helpers.tsx

```diff
@@ -0,0 +1,205 @@
+  Action,
+  confirmAlert,
+  getPreferenceValues,
+  open,
+  openCommandPreferences,
+} from "@raycast/api";
+import {
+  addToCache,
+  checkIfCached,
+  getFromCache,
+  RECENTLY_USED,
+} from "../../cache";
+import { Service } from "../../util/service";
+import { compareByDate, compareByName, toId } from "../../util/compare";
+import { AegisDB } from "../../util/AegisDB";
+
+const {
+  excludeNames: excludeNamesCsv = "",
+  primaryActionIsCopy,
+  recentlyUsedOrder,
+  aegisDbPath,
+  aegisPassword,
+} = getPreferenceValues<{
+  excludeNames: string;
+  primaryActionIsCopy: boolean;
+  recentlyUsedOrder: boolean;
+  aegisDbPath: string;
+  aegisPassword: string;
+}>();
+
+/**
+ * Placeholder for Services with broken seeds
+ */
+export const CORRUPTED = "corrupted";
+
+/**
+ * Just a type to use in other files
+ */
+export type setItemsFunction = (
+  value:
+    | ((prevState: { otpList: Service[]; isLoading: boolean }) => {
+        otpList: Service[];
+        isLoading: boolean;
+      })
+    | { otpList: Service[]; isLoading: boolean }
+) => void;
+
+/**
+ * Loads OTP services to show, sort and exclude values from extension propertiesz
+ */
+export async function loadData(setItems: setItemsFunction): Promise<void> {
+  // load Aegis
+  const aegisDB = new AegisDB(aegisDbPath);
+  try {
+    //Load the database
+    await aegisDB.loadDb();
+
+    //Decrypt the database
+    const db_entries = await aegisDB.decryptDb(aegisPassword);
+    const otpServices: Service[] = db_entries.map((i) => {
+      return {
+        id: i.uuid,
+        name: i.name,
+        digits: i.info.digits,
+        period: i.info.period,
+        seed: i.info.secret,
+        accountType: "service",
+        issuer: i.issuer,
+        // logo: i.logo,
+        logo: "",
+        type: "service",
+      };
+    });
+
+    setItems({
+      otpList: await sortServices(otpServices),
+      isLoading: false,
+    });
+  } catch (error) {
+    setItems({
+      otpList: [],
+      isLoading: false,
+    });
+  }
+}
+
+/**
+ *
+ */
+export async function checkError(otpList: Service[], isLoading: boolean) {
+  // if (otpList.length === 0) return;
+  if (isLoading) return;
+
+  // filter out all the corrupted otp
+  const all = otpList.filter(
+    (otp) => otp.type == "service" && otp.seed != null
+  );
+  // if none of the otp are valid assume there is a problem with the password
+  if (all.length === 0) {
+    confirmAlert({
+      title: "No valid OTP",
+      message: "Check your Aegis Encryption Password in settings",
+      primaryAction: {
+        title: "Open Preferences",
+        onAction: () => openCommandPreferences(),
+      },
+      dismissAction: {
+        title: "Cancel",
+        onAction: () =>
+          confirmAlert({
+            title: "No valid OTP",
+            message: "Check Q&A in store",
+            primaryAction: {
+              title: "Open Store Page",
+              onAction: () => open("https://www.raycast.com/guga4ka/authy"),
```

Comment:
This should be corrected 🙂 Maybe it should go to preferences?

Reply:
Made primary action go to preferences and removed the secondary action entirely in [84b4f36](https://github.com/raycast/extensions/pull/13860/commits/84b4f36d4ec3a19b066ceadbf137bb1ff072bf45)